source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,246 @@
+#include "render/RenderManager.h"
+#include <sstream>
+#include <algorithm>
+
+//! Values used to know whether to show and/or hide a mesh
+
+CullingManager::CullingManager(CameraManager* cameraManager):
+    mFirstIter(false),
+    mCm(cameraManager),
+    mCullTilesFlag(false)
+{
+    mActivePlanes =(Ogre::Plane(0, 0, 1, 0));
+    // init the Ogre vector
+    for (unsigned int i = 0; i < 8; ++i)
+    {
+        mOgreVectorsArray[i].x = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].y = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].z = static_cast<Ogre::Real>(0.0);
+    }
+}
+
+void CullingManager::cullTiles()
+{
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(SHOW | HIDE);
+}
+
+void CullingManager::startTileCulling()
+{
+    getIntersectionPoints();
+
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+    mOldWalk = mWalk;
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+
+void CullingManager::stopTileCulling()
+{
+    mCullTilesFlag = false;
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());        
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrDetachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::showAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            RenderManager::getSingleton().rrAttachEntity(gm->getTile(ii, jj));
+        }
+    }
+}
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
```

Comment:
Note that here, you are using mode as an array of bytes. using a int64_t is not usefull for an array that can have only 2 values. Moreover, for this kind of usage, we usually use an unsigned int such as uint32_t.
Since there are still changes to do, it would be nice to change that at the same time.